### ISSUE TITLE
SSC: Train/inference distribution mismatch — training on Rust code, inference on shell (KAIZEN-103)

### ISSUE THREAD

**Issue by noahgift** [paiml/bashrs#172]
## Problem

The CodeBERT linear probe is trained on **Rust code** embeddings (corpus entry `.input` field), but at inference time (`classify_with_probe`), users provide **shell scripts**. This is a distribution mismatch.

### Root Cause

`corpus_baseline_entries()` in `baselines.rs` line 94:
```rust
(e.input.clone(), row.label)  // e.input is Rust code, not shell
```

The Rust code is the transpiler input. The transpiled shell output exists in `DatasetRow.actual_output` but isn't used for embedding extraction.

### Impact

- MCC=0.321 is measured on Rust code test set — actual shell script performance is unknown
- Adversarial shell script augmentation (`--augment`) produces domain mismatch (worse MCC)
- The probe may not generalize to shell scripts at all

### Proposed Fix

Change `corpus_baseline_entries()` to use transpiled shell output instead of Rust input:

```rust
// Current (wrong):
(e.input.clone(), row.label)  // Rust code

// Fixed:
(row.actual_output.clone(), row.label)  // Transpiled shell
```

This requires access to `DatasetRow.actual_output`, which means running transpilation on each entry during export. The transpilation is already run for V2 scoring, so this just needs to be wired through.

### Acceptance Criteria

- [ ] Training data uses transpiled shell output, not Rust input
- [ ] MCC measured on shell script embeddings
- [ ] Adversarial shell script augmentation works (same domain)
- [ ] `bashrs classify --probe` inference matches training distribution

### References

- #171 (data labeling gap — related but separate issue)
- classifier-pipeline-v1.yaml
- SSC v11 spec Section 4.3
